wordpress-mobile/WordPress-iOS
# Tenor integration

This PR solves the issue of adding Gif support using Google's Tenor API ( #12997 )

I've created a simple wrapper to replace the Giphy object in the Giphy's old SDK. Since the response we get from the Tenor's API has a completely different structure, and there isn't 1-to-1 mapping between all the fields, I had to select suitable replacements among the fields that are available in the Tenor's API response. They might not be the best choices yet, that's why I haven't simplified the TenorMedia class further. I left some room for some experimentation if needed.

I've tried to use the existing Giphy related classes as much as possible. Therefore, there's a a lot of renaming and some relocating involved. That's why the PR is a bit long.

There's some force unwrapping and some unnecessary casting/bridging I'm gonna fix, but I wanted to get some feedback before doing so.

**To test:**

- Create an ApiKey by following the instructions in the [documentation](https://tenor.com/gifapi/documentation#quickstart).
- Add the key into the credentials file `.configure-files/wpcom_app_credentials` 

```
WPCOM_APP_ID=....
WPCOM_APP_SECRET=....
TENOR_APP_ID={The key you've just created}
```

- Rake dependencies
- Build and run, check it builds okay
- In the Media Library, tap + and select Tenor from the available options
- In Aztec, tap the + and then ... on the format bar and and select Tenor from the available options
